# Chisquare Test

## Hypotheses

Null Hypothesis(H0): There is no significant association between the type of device purchased (Smart Thermostat or Smart Light) and the customer's satisfaction level. This implies that satisfaction is independent of the device type.

Alternative Hypothesis(Ha): There is a significant association between the type of device purchased and the customer's satisfaction level. This implies that satisfaction depends on the device type.



In [1]:
import numpy as np
from scipy.stats import chi2,chi2_contingency

In [2]:
data=np.array([[50,70],[80,100],[60,90],[30,50],[50,20]])

In [3]:
# Perform the Chi-Square test for independence
chi2_statistic, p_value, dof, expected = chi2_contingency(data)
print(chi2_statistic)
print(p_value)
print(dof)

23.649190315856984
9.390467988105083e-05
4


In [4]:
# Determine critical value for 0.05 significance level with the calculated degrees of freedom
alpha = 0.05
dof = 4
critical_value = chi2.ppf(1 - alpha, dof)
critical_value

9.487729036781154

##### Decision: Since the critical value is greater than chi2_statistic(5.638<9.487), failed to reject the null hypothesis

# Hypothesis Testing

### Hypotheses

Null Hypothesis(H0): The mean weekly operating cost is equal to the theoretical cost model prediction.
    H0: mu=1000+5(600)=4000

Alternative Hypothesis(Ha): The mean weekly operating cost is greater than the theoretical cost model prediction.
    Ha: mu>4000

In [5]:
import numpy as np
from scipy.stats import norm

In [6]:
# Given Data
x_bar = 3050  # Sample mean weekly cost
mu = 4000     # Theoretical mean weekly cost
sigma = 5*25  # Standard deviation
n = 25         # Sample size(25 restaurants)

In [7]:
# Calculate the test statistic (t)
t_statistic = (x_bar-mu) / (sigma/np.sqrt(n))
t_statistic

-38.0

In [8]:
# Determine the critical value for alpha = 0.05 (one-tailed test)
alpha = 0.05
z_critical = norm.ppf(1-alpha)
z_critical

1.6448536269514722

In [9]:
t_statistic, z_critical

(-38.0, 1.6448536269514722)

#### Decision: Since the t_statistic is less than Z_critical value(-38<1.645), Failed to reject the Null Hypothesis

#### Conclusion

Based on the calculated test statistic and the comparison with the critical value, there is not enough evidence to support the restaurant owners' claim that the weekly operating costs are higher than the model suggests